actor/v7action/service_instance_sharing_test.go

```diff
@@ -251,5 +163,115 @@ var _ = Describe("Service Instance Sharing", func() {
+			}
+		})
+
+		JustBeforeEach(func() {
+			warnings, executionError = actor.ShareServiceInstanceToSpaceAndOrg(
+				serviceInstanceName,
+				targetedSpaceGUID,
+				targetedOrgGUID,
+				serviceInstanceSharingParams,
+			)
+		})
+
+		itValidatesParameters()
+
+		When("we make a successful share request", func() {
+			spaceGUID := "fake-space-guid"
+			serviceInstanceGUID := "fake-service-instance-guid"
+
+			BeforeEach(func() {
+				fakeCloudControllerClient.GetSpacesReturns(
+					[]resources.Space{{GUID: spaceGUID}},
+					ccv3.IncludedResources{},
+					ccv3.Warnings{"some-space-warning"},
+					nil,
+				)
+
+				fakeCloudControllerClient.GetServiceInstanceByNameAndSpaceReturns(
+					resources.ServiceInstance{GUID: serviceInstanceGUID},
+					ccv3.IncludedResources{},
+					ccv3.Warnings{"some-service-instance-warning"},
+					nil,
+				)
+			})
+
+			It("makes a request to the cloud controller", func() {
+				Expect(fakeCloudControllerClient.GetServiceInstanceByNameAndSpaceCallCount()).To(Equal(1))
+				Expect(fakeCloudControllerClient.GetSpacesCallCount()).To(Equal(1))
+				Expect(fakeCloudControllerClient.ShareServiceInstanceToSpacesCallCount()).To(Equal(1))
+
+				actualServiceInstanceGUID, actualSpaces := fakeCloudControllerClient.ShareServiceInstanceToSpacesArgsForCall(0)
+				Expect(actualServiceInstanceGUID).To(Equal(serviceInstanceGUID))
+				Expect(actualSpaces[0]).To(Equal(spaceGUID))
+
+				Expect(executionError).To(BeNil())
+				Expect(warnings).To(ConsistOf("some-space-warning", "some-service-instance-warning"))
+			})
+		})
+
+		When("sharing request returns an error", func() {
+			spaceGUID := "fake-space-guid"
+			serviceInstanceGUID := "fake-service-instance-guid"
+
+			BeforeEach(func() {
+				fakeCloudControllerClient.GetSpacesReturns(
+					[]resources.Space{{GUID: spaceGUID}},
+					ccv3.IncludedResources{},
+					ccv3.Warnings{"some-space-warning"},
+					nil,
+				)
+				fakeCloudControllerClient.GetServiceInstanceByNameAndSpaceReturns(
+					resources.ServiceInstance{GUID: serviceInstanceGUID},
+					ccv3.IncludedResources{},
+					ccv3.Warnings{"some-service-instance-warning"},
+					nil,
+				)
+				fakeCloudControllerClient.ShareServiceInstanceToSpacesReturns(
+					resources.RelationshipList{},
+					ccv3.Warnings{"some-share-warning"},
+					errors.New("cannot share the instance"),
+				)
+			})
+
+			It("returns an error and warnings", func() {
+				Expect(executionError.Error()).To(Equal("cannot share the instance"))
+				Expect(warnings).To(ConsistOf("some-space-warning", "some-service-instance-warning", "some-share-warning"))
+			})
+		})
+	})
+
+	Describe("UnShareServiceInstanceToSpaceAndOrg", func() {
+		BeforeEach(func() {
+			serviceInstanceSharingParams = ServiceInstanceSharingParams{
+				SpaceName: shareToSpaceName,
+				OrgName:   types.OptionalString{},
+			}
+		})
+
+		JustBeforeEach(func() {
+			warnings, executionError = actor.ShareServiceInstanceToSpaceAndOrg(
```

Comment:
Should this be `UnshareServiceInstanceToSpaceAndOrg`?

Reply:
panic()